### PR TITLE
Apply eslint, prettier, and black

### PR DESCRIPTION
**Pull request by Farkites**
## About
 
Applied format and lint tools

### Added
* `prettier` and `black` lint/format scripts have been added.
* React 18 job has been added to circleCI workflow.

### Changed
* Dash has been upgraded to 2.* in requirements.txt and tests/requirements.txt.
* Dependencies have been upgraded to lastest versions.
* CircleCI workflow has been updated to version 2.1.
* Python version in CircleCI workflow has been updated to 3.11.
* Node version in CircleCI workflow has been updated to 20.5.1.

### Fixed
* CircleCI workflow has been fixed. 

## Pre-Merge checklist
- [x] The project was correctly built with `npm run build:all`.
- [x] If there was any conflict, it was solved correctly.
- [x] All changes were documented in CHANGELOG.md.
- [x] All tests on CircleCI have passed.
- [x] All Percy visual changes have been approved.
- [ ] Two people have :dancer:'d the pull request. You can be one of these people if you are a Dash Cytoscape core contributor.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "build:js-dev-extra": "webpack --config webpack.dev.extra.config.js",
     "build:js-extra": "webpack --config webpack.prod.extra.config.js",
     "build:js-all": "npm run build:js && npm run build:js-dev && npm run build:js-dev-extra && npm run build:js-extra",
-    "build:backends": "dash-generate-components ./src/lib/components dash_cytoscape --r-prefix 'cyto' --jl-prefix 'cyto'",
+    "build:backends": "dash-generate-components ./src/lib/components dash_cytoscape --r-prefix 'cyto' --jl-prefix 'cyto' && npm run private::format.black",
```
Comment (alexcjohnson) [inline package.json:28]: You probably want to only format the build artifacts here, not the rest of the Python files, otherwise you can run into problems if you rebuild while editing something else.

File: .circleci/config.yml [hunk 2]
```diff
@@ -35,40 +50,46 @@ jobs:
                 name: Run eslint
                 command: ./node_modules/.bin/eslint src
                 when: always
+            
+            - run:
+                name: Run prettier
+                command: |
+                    ./node_modules/.bin/prettier --config .prettierrc --write src/**/*.js src/**/**/*.js
```
Comment (alexcjohnson) [inline .circleci/config.yml:57]:
```suggestion
                    npm run lint:lint.prettier
```
🌴, also this is the `format` version, you want the `lint` version right? Anyway it's better if we can have CI run commands from package.json, so that we know it's doing the same thing a dev will do when they're writing their code.

That said those command names `lint:lint.prettier` and `lint:format.prettier` and such are awkward. `lint...` commands should just test, `format...` should edit the files. So you could just call these `lint.prettier` and `format.prettier`, or follow the convention we have in Dash (examples from DCC but all the pieces basically do the same):
- [`lint`](https://github.com/plotly/dash/blob/6eaf2e17c25f7ca1847c41aafeb18e87c586cb9f/components/dash-core-components/package.json#L32) runs all the linting subcommands, [`format`](https://github.com/plotly/dash/blob/6eaf2e17c25f7ca1847c41aafeb18e87c586cb9f/components/dash-core-components/package.json#L31) runs all the formatting subcommands
- The [subcommands](https://github.com/plotly/dash/blob/6eaf2e17c25f7ca1847c41aafeb18e87c586cb9f/components/dash-core-components/package.json#L15-L21) are all called `private::lint.<tool>` and `private::format.<tool>`

File: .circleci/config.yml [hunk 3]
```diff
@@ -35,40 +50,46 @@ jobs:
...
 
         steps:
             - checkout
+            - browser-tools/install-chrome
+            - browser-tools/install-chromedriver
 
             - restore_cache:
                 key: deps1-{{ .Branch }}-{{ checksum "tests/requirements.txt" }}
-
+
```
Comment (alexcjohnson) [inline .circleci/config.yml:74]: Does your code editor have a "strip trailing spaces on save" option?

File: .circleci/config.yml [hunk 4]
```diff
@@ -90,52 +111,58 @@ jobs:
                 name: Integration Tests - Usage Apps Rendering
                 command: |
                     . venv/bin/activate
-                    pytest --nopercyfinalize --headless tests/test_usage.py
+                    python -m pytest --nopercyfinalize --headless tests/test_usage.py
```
Comment (alexcjohnson) [inline .circleci/config.yml:114]: Curious why this change was necessary? `pytest` was finding the one in the base environment rather than the venv or something? We haven't needed to add `python -m` in front of other similar commands

File: .circleci/config.yml [hunk 5]
```diff
@@ -90,52 +111,58 @@ jobs:
...
-
-workflows:
-    version: 2
+workflows:    
+    version: 2.1
     build:
         jobs:
-            - "python-3.6":
+            - python-3-11:
                 context: dash-docker-hub
```
Comment (alexcjohnson) [inline .circleci/config.yml:150]: not new in this PR, but isn't this `context` redundant with the `auth` you already have defined in each executor?

File: .pylintrc [hunk 6]
```diff
@@ -460,6 +453,8 @@ init-import=no
 # builtins.
 redefining-builtins-modules=six.moves,past.builtins,future.builtins
 
+disable=W0640
```
Comment (alexcjohnson) [inline .pylintrc:456]: I haven't seen the need to disable this one before, but it looks like a potentially [dangerous situation](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/cell-var-from-loop.html) so better if we can keep it!

File: _validate_init.py [hunk 7]
```diff
@@ -9,32 +9,32 @@
 import importlib
 
 
-components_package = 'dash_cytoscape'
+components_package = "dash_cytoscape"
```
Comment (alexcjohnson) [inline _validate_init.py:12]: FWIW in other packages we've mostly deleted this file and the associated package script `validate-init` - we deprecated it [4 years ago](https://github.com/plotly/dash-cytoscape/blob/master/CHANGELOG.md#changed-3) so I think we can 🔪 here 😅

File: demos/usage-animated-bfs.py [hunk 8]
```diff
@@ -16,56 +16,56 @@
 app.css.config.serve_locally = True
```
Comment (alexcjohnson) [inline demos/usage-animated-bfs.py:16]: You can do this in a later PR if you want but just so we don't forget: all these `serve_locally` lines can be removed (has been the default since Dash v1.0!) and the imports (`html`, `dcc`, etc) should be [updated to Dash 2.x](https://dash.plotly.com/dash-2-0-migration#recommended-updates-%E2%80%94-import-statements)

File: setup.py [hunk 9]
```diff
@@ -4,43 +4,43 @@
...
+        "Intended Audience :: Developers",
+        "Intended Audience :: Education",
+        "Intended Audience :: Financial and Insurance Industry",
+        "Intended Audience :: Healthcare Industry",
+        "Intended Audience :: Manufacturing",
+        "Intended Audience :: Science/Research",
+        "License :: OSI Approved :: MIT License",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.5",
```
Comment (alexcjohnson) [inline setup.py:39]: remove 3.5, add up to 3.11 to classifiers, and add a `python_requires=">=3.6"` line

File: src/demo/App.js [hunk 10]
```diff
@@ -2,73 +2,88 @@
 import React from 'react';
 import Cytoscape from '../lib/components/Cytoscape.react.js';
 
-
 const elements = [
     {
         data: {id: 'one', label: 'Node 1'},
-        position: {x: 50, y: 50}
+        position: {x: 50, y: 50},
```
Comment (alexcjohnson) [inline src/demo/App.js:8]: you can actually just delete `src/demo` entirely, along with the `start` package script and the `webpack-serve` devDep. AFAICT nobody uses the standalone react demo anymore even during early component dev, we just dev in an actual Dash app. So likely as not the standalone demo is out-of-date or outright broken in one way or another.

File: usage-elements.py [hunk 11]
```diff
@@ -139,152 +118,165 @@
...
+        ),
+    ]
+)
 
 
 # ############################## CALLBACKS ####################################
-@app.callback(Output('tap-node-json-output', 'children'),
-              [Input('cytoscape', 'tapNode')])
+@app.callback(
+    Output("tap-node-json-output", "children"), [Input("cytoscape", "tapNode")]
```
Comment (alexcjohnson) [inline usage-elements.py:212]:
```suggestion
    Output("tap-node-json-output", "children"), Input("cytoscape", "tapNode")
```
We can modernize all the callback definitions

File: demos/usage-animated-bfs.py (hunk 8 above)
Comment (alexcjohnson) [reply demos/usage-animated-bfs.py:16]: nonblocking, but the apps in the `demos` folder still have `serve_locally = True`

File: .circleci/config.yml (hunk 4 above)
Comment (alexcjohnson) [reply .circleci/config.yml]: nonblocking, but I'm still curious about this change 😸

**Review by alexcjohnson — APPROVED** [review]
💃 Great work!

File: demos/usage-animated-bfs.py (hunk 8 above)
Comment (Farkites) [reply demos/usage-animated-bfs.py:16]: My bad, I'll remove them

File: .circleci/config.yml (hunk 4 above)
Comment (Farkites) [reply .circleci/config.yml]: It is actually not needed. I added it because at some point circleci was failing to find `pytest` and this seemed to solve it. I just removed it.

**Comment by emilykl** [comment]
🥳 🥳 🥳 

This rocks! Great work @Farkites !
